TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/TrafficReplayer.java

```diff
@@ -376,79 +383,83 @@ void runReplayWithIOStreams(Duration observedPacketConnectionTimeout,
-            DiagnosticTrackableCompletableFuture<String, TransformedTargetRequestAndResponse>> requestFutureMap) {
+    getRecordedRequestReconstructCompleteHandler(ReplayEngine replayEngine) {
         return (requestKey, request) -> {
+            replayEngine.setFirstTimestamp(request.getFirstPacketTimestamp());
+
             var requestPushFuture = transformAndSendRequest(replayEngine, request, requestKey);
-            requestFutureMap.put(request, requestPushFuture);
+            requestFutureMap.put(requestKey, requestPushFuture);
             requestPushFuture.map(f->f.whenComplete((v,t)-> log.atTrace()
                     .setMessage(()->"Summary response value for " + requestKey + " returned=" + v).log()),
                     ()->"logging summary");
         };
     }
 
-    private static Consumer<RequestResponsePacketPair>
+    private final static Instant startTime = Instant.now();
+
+    private Consumer<RequestResponsePacketPair>
     getRecordedRequestAndResponseReconstructCompleteHandler(
             AtomicInteger successCount,
             AtomicInteger exceptionCount,
-            SourceTargetCaptureTuple.TupleToFileWriter tupleWriter,
-            ConcurrentHashMap<HttpMessageAndTimestamp, DiagnosticTrackableCompletableFuture<String, TransformedTargetRequestAndResponse>>
-                    responseInProgressMap,
-            ConcurrentHashMap<HttpMessageAndTimestamp, DiagnosticTrackableCompletableFuture<String, TransformedTargetRequestAndResponse>>
-                    targetTransactionInProgressMap) {
+            SourceTargetCaptureTuple.TupleToFileWriter tupleWriter) {
         return rrPair -> {
             log.atTrace().setMessage(()->"Done receiving captured stream for this " + rrPair.requestData).log();
-            var requestData = rrPair.requestData;
-            var resultantCf = responseInProgressMap.remove(requestData)
+            var resultantCf = requestFutureMap.remove(rrPair.requestKey)
                     .map(f ->
                             f.handle((summary, t) -> {
+                                TransformedTargetRequestAndResponse rval = null;
                                 try {
-                                    TransformedTargetRequestAndResponse rval =
-                                            packageAndWriteResponse(tupleWriter, rrPair, summary, t);
+                                    rval = packageAndWriteResponse(tupleWriter, rrPair, summary, t);
                                     successCount.incrementAndGet();
                                     return rval;
                                 } catch (Exception e) {
-                                    log.atInfo().setCause(e).setMessage("Exception - base64 gzipped traffic stream: " +
-                                            Utils.packetsToCompressedTrafficStream(rrPair.requestData.stream())).log();
+                                    log.atInfo().setCause(e).setMessage("Exception for request {}: ")
```

Comment:
Does this update mean that this exception is _not_ related to a base64 gzipped traffic stream? If it still is, I think that's useful context to include.